packages/react-core/src/hooks/useIsSignedIn.ts

```diff
@@ -0,0 +1,46 @@
+import { getCurrentUser } from 'aws-amplify/auth';
+import { Hub, HubCallback } from '@aws-amplify/core';
+import { useEffect, useState } from 'react';
+import useDataState, { DataState } from './useDataState';
+
+const action = async (_: { isSignedIn: boolean }, __: undefined) => {
+  await getCurrentUser();
+  return { isSignedIn: true };
```

Comment:
Do we need to catch here and return the `defaultState`? I know the `defaultState` is set initially but I'm more concerned about an uncaught error here when user is not signed in

Reply:
The idea is that the uncaught error will be reflected in the dataState:

const expectedErrorState = {
    data: { isSignedIn: false },
    hasError: true,
    isLoading: false,
    message: 'Authorization error',
};

Reply:
I might be wrong but doesn't `getCurrentUser()` always throw if the user is not signed in? And in that case it's not really an error scenario for the customer